docs/developer/architecture/coding-principles.md

```diff
@@ -26,27 +26,67 @@
 ## III. Simplicity
 
 1. Avoid layers of indirection when they are not needed (e.g. "pass-through methods").
-
-### IV. Testing
+2. Avoid needlessly wrapping objects, especially primitive datatypes.
+
+### IV. General Coding Style
+
+1. Use `var` instead of explicit types (helps with clarity)
+2. Avoid `final` in method args and local variables
+3. Use `final` in field declarations
+4. Avoid `static` fields except in constants or when absolutely necessary. (you should be able to provide a reason).
+5. Use interfaces to define shared constants
+6. Use "minimally required types" (or "smallest possible API"), e.g. use `ObjectMapper` instead of `TypeManager`
+   , or use a `String` instead of a more complex object containing the String, etc.
+7. Use either `public` members, which are documented and tested, or `private` members.
+8. Avoid package-private members, especially if only needed for testing
+9. Avoid `protected` members unless they're intended to be overridden.
+10. Use package-private classes if they're not needed outside the package, e.g. implementation classes
+11. Avoid using `enum`s for anything other than named integer enumerations.
+12. Avoid using static classes as much as possible. Exceptions to this are helper functions and test utils, etc. as well as static inner classes.
+13. Use only camel case and no prefixes for naming.
+14. Avoid unnecessary `this.` (exception: in builders)
```

Comment:
I would avoid them in builders as well, I mean, as long as there isn't a name overlapping (in that case the `this` will be necessary) I would avoid it.

Reply:
that was what I meant. I'll make it explicit though